platform/linux-generic/odp_schedule_basic.c

```diff
@@ -1430,7 +1420,11 @@ static inline int do_schedule(odp_queue_t *out_q, odp_event_t out_ev[], uint32_t
 		sched_local.grp_epoch = epoch;
 	}
 
-	first_id = sched_local.grp_weight[grp_round];
+	if (odp_unlikely(num_grp == 0))
+		return 0;
+
+	first_id = sched_local.grp_idx;
+	sched_local.grp_idx = (first_id + 1) % num_grp;
```

Comment:
> Both methods round-robin the first group index the same way.

Not quite, as the old code had some imbalance when the number of groups did not divide GRP_WEIGHT_TBL_SIZE evenly. So from that point-of-view this code seems better.

I suppose the rationale for the old code was to avoid the non-constant and not-power-of-two modulo operation that this code now has. So this change trades some cycles needed by the modulo opration to some memory footprint savings. Perhaps the commit message could be updated a bit.

Reply:
Commit message updated in v2